scripts/generator/graphql_generator/csharp/SDK/CheckoutMutator.cs.erb

```diff
@@ -0,0 +1,268 @@
+            Action<Payment> pollPayment = (payment) => {
+                PaymentPoller.PollPaymentReady(payment.id(), (Payment newPayment, ShopifyError error) => {
+                    if (error != null) {
+                        callback(error);
+                        return;
+                    } 
+
+                    DidMutateCheckout(payment.checkout(), null);
+                    callback(null);
+                });
+            };
+
+            Action checkoutWithTokenizedPayment = () => {
+                MutationQuery query = new MutationQuery();
+                DefaultQueries.checkout.CheckoutCompleteWithTokenizedPayment(query, checkout.id(), tokenizedPaymentInput);
+
+                Client.Mutation(query, (Mutation response, ShopifyError error) => {
+                    if (error != null) {
+                        callback(error);
+                        return;
+                    } 
+
+                    var responseNode = response.checkoutCompleteWithTokenizedPayment();
+                    var payment = responseNode.payment();
+                    var userErrors = responseNode.userErrors();
+
+                    DidMutateCheckout(responseNode.checkout(), userErrors);
+
+                    if (userErrors.Count > 0) {
+                        HandleUserError(callback);
+                        return;
+                    }
+
+                    if (payment.ready()) {
+                        callback(null);
+                    } else {
+                        pollPayment(payment);
+                    }
+                });
+            };
+
+            // Ensure we can checkout first
+            if (checkout.ready()){
+                checkoutWithTokenizedPayment();
+                return;
+            } 
+
+            CheckoutPoller.PollCheckoutReady(checkout.id(), (Checkout polledCheckout, ShopifyError error) => {
+                if (error != null) {
+                    callback(error);
+                    return;
+                } 
+
+                checkoutWithTokenizedPayment();
+            });
+        }
+
+        private void HandleCheckoutMutation(
+            Checkout newCheckout, 
+            List<UserError> userErrors, 
+            CheckoutPoll pollFinishedHandler,
+            CompletionCallback callback
+        ) {
+            DidMutateCheckout(newCheckout, userErrors);
+
+            if (userErrors.Count > 0) {
+                HandleUserError(callback);
+                return;
+            } 
+
+            if (newCheckout.ready()) {
+                callback(null);
+            } else {
+                CheckoutPoller.PollCheckoutAndUpdate(newCheckout.id(), pollFinishedHandler, callback);
+            }
+        }
+
+        private void HandleUserError(CompletionCallback callback) {
+            ShopifyError error = new ShopifyError(
+                ShopifyError.ErrorType.UserError,
+                "There were issues with some of the fields sent. See `cart.UserErrors`"
```

Comment:
We're no longer in `cart` so this description needs to change.  I wonder if this would be better if it lived in `Cart`, maybe a delegate methods: `ErrorForMutateCheckoutUserErrors(UserErrors) -> ShopifyError` for error  and `DidMutateCheckout(newCheckout)` for success